Lib/test/test_wsgiref.py

```diff
@@ -193,6 +193,23 @@ def bad_app(environ, start_response):
                 ))
                 self.assertEqual(err.splitlines()[-2], exc_message)
 
+    @unittest.skipIf(support.python_is_optimized(),
+                     "Python was compiled with optimizations")
+    def test_hop_by_hop_validation_error(self):
+        def bad_app(environ, start_response):
+            start_response("200 OK", [('Content-Type', 'text/plain'),
+                                      ('Connection', 'close')])
+            return ["Hello, world!"]
+        out, err = run_amock(bad_app)
+        self.assertTrue(out.endswith(
+            b"A server error occurred.  Please contact the administrator."
+        ))
+        self.assertEqual(
+            err.splitlines()[-2],
+            "AssertionError: Hop-by-hop header, 'Connection: close',"
+            " not allowed"
+        )
+
```

Comment:
Why don't simply do `.. assertRaise(AssertionError ...`?

Reply:
@eamanu Thanks for the review and the suggestion.  I didn't do this originally to keep it consistent with the other tests (test_simple_validation_error and test_status_validation_errors).  But, based on Raymond's review and your comment, I've changed it to just check that an `AssertionError` is raised.

Reply:
IMO the assertRaise is poorly written